### PR TITLE
feat: create `supportsLSPInterface` function

### PR DESCRIPTION
**Pull request by b00ste**
# What does this PR introduce?

This PR creates a new function that would help identifying LSP contract. Function taken from `erc725.js` library.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "dependencies": {
         "@erc725/erc725.js": "^0.20.0",
         "@lukso/lsp-smart-contracts": "^0.11.1",
```
Comment (CJ42) [inline package.json]: Would be good to update the version of the lsp-smart-contracts package to the latest

File: package.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
...
+        "@openzeppelin/contracts": "^5.0.1",
```
Comment (CJ42) [inline package.json]: I suggest we keep the same version as the one used in `@lukso/lsp-smart-contracts`, which is 4.9.2

```suggestion
        "@openzeppelin/contracts": "^4.9.2",
```

File: src/helpers/supportsLSPInterface/supportsLSPInterface.test.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        },
+    };
+    const mockProvider = {} as Provider;
+
+    it('it should support LSP0 interface and return true', async () => {
+        expect(await supportsLSPInterface(INTERFACE_IDS.LSP0ERC725Account, mockERC165 as ERC165)).to
+            .be.true;
+    });
+
+    it("it shouldn't support LSP0 interface and return false", async () => {
```
Comment (CJ42) [inline src/helpers/supportsLSPInterface/supportsLSPInterface.test.ts]: I am confused by the name of this test. It says `shouldn't support LSP0 interface`, but the interface ID used is LSP7

File: src/helpers/supportsLSPInterface/supportsLSPInterface.test.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        const contractAddress = '0xcafe';
+
+        await expect(
+            supportsLSPInterface(INTERFACE_IDS.LSP7DigitalAsset, contractAddress, mockProvider),
+        ).to.be.rejectedWith(
+            `The parameter \`contract\` is not a valid address nor a valid contract instance of ERC165 contract. Value: '${contractAddress}'`,
+        );
+    });
+
+    it('should revert if contract is not an addressable contract', async () => {
```
Comment (CJ42) [inline src/helpers/supportsLSPInterface/supportsLSPInterface.test.ts:51]: What is meant by `not an addressable`?

File: src/helpers/supportsLSPInterface/supportsLSPInterface.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+): Promise<boolean> {
+    if (!isAddress(contract) && !isAddressable(contract)) {
+        throw new Error(
+            `The parameter \`contract\` is not a valid address nor a valid contract instance of ERC165 contract. Value: '${contract}'`,
+        );
+    }
+
+    let contractInstance: ERC165;
+
+    if (isAddress(contract)) {
```
Comment (CJ42) [inline src/helpers/supportsLSPInterface/supportsLSPInterface.ts:22]: If the two conditionals line 14 are skipped, it means that this line will be sure to pass.

**Review by CJ42 — COMMENTED** [review]
Added review comments

File: src/helpers/supportsLSPInterface/supportsLSPInterface.ts (hunk 5 above)
Comment (b00ste) [reply src/helpers/supportsLSPInterface/supportsLSPInterface.ts:22]: Yep, here I need to know which of those condition is true

File: src/helpers/supportsLSPInterface/supportsLSPInterface.test.ts (hunk 4 above)
Comment (b00ste) [reply src/helpers/supportsLSPInterface/supportsLSPInterface.test.ts:51]: it is a ethers check, is addressable means its an object that has the async `getAddress()` function, a contract is addressable, a signer is addressable

**Review by CJ42 — APPROVED** [review]

**Pull request merged**